backend/app/main.py

```
@@ -0,0 +1,136 @@
+        if user_data is None:
+            raise HTTPException(status_code=404, detail="User not found")
+        
+        return {
+            "success": True,
+            "user_id": user_id,
+            "comparisons": user_data.comparison_history or []
+        }
+        
+    except Exception as e:
+        logger.error(f"Error fetching comparisons for user {user_id}: {e}")
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)